### PR TITLE
🐛 CABPK should not log the token upon renewal

### LINKED ISSUE #3771: Kubeadm bootstrap provider is logging tokens
**What steps did you take and what happened:**
Creating a machine with the Kubeadm Bootstrap Provider results in tokens getting logged

```
"msg"="Altering JoinConfiguration.Discovery.BootstrapToken" "name"="bmo-md-0-ppxgx" "namespace"="default" "reconcilerGroup"="bootstrap.cluster.x-k8s.io" "reconcilerKind"="KubeadmConfig" "Token"="rqn8yd.hq099n1e8uu6v2vz"
```

**What did you expect to happen:**
No tokens should be logged, since they are secrets

/kind bug
/area bootstrap

### PR DESCRIPTION
**Pull request by dswij**
**What this PR does / why we need it**:
Token should be secret and not logged

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3771

### CODE REVIEW

File: bootstrap/kubeadm/controllers/kubeadmconfig_controller.go [hunk 1]
```diff
@@ -735,7 +735,6 @@ func (r *KubeadmConfigReconciler) reconcileDiscovery(ctx context.Context, cluste
 		}
 
 		config.Spec.JoinConfiguration.Discovery.BootstrapToken.Token = token
-		log.Info("Altering JoinConfiguration.Discovery.BootstrapToken", "Token", token)
```
Comment (vincepri) [inline bootstrap/kubeadm/controllers/kubeadmconfig_controller.go:738]: Instead of removing the whole line, should we just remove the token instead?

Comment (dswij) [reply bootstrap/kubeadm/controllers/kubeadmconfig_controller.go:738]: My bad, let me fix that

**Review by vincepri — APPROVED** [review]
Mind squashing commits?

/approve

**Comment by dswij** [comment]
> 
> 
> Mind squashing commits?

no problem

File: bootstrap/kubeadm/controllers/kubeadmconfig_controller.go [hunk 2]
```diff
@@ -735,7 +735,7 @@ func (r *KubeadmConfigReconciler) reconcileDiscovery(ctx context.Context, cluste
...
+		log.Info("Altering JoinConfiguration.Discovery.BootstrapToken", "Token")
```
Comment (benmoss) [inline bootstrap/kubeadm/controllers/kubeadmconfig_controller.go:738]: The arguments to `Info` are supposed to be the message followed by key/value pairs, so just having `"Token"` isn't right.
```suggestion
		log.Info("Altering JoinConfiguration.Discovery.BootstrapToken")
```

**Comment by benmoss** [comment]
> > Mind squashing commits?
> 
> no problem.
> 
> Though, i'm not sure what u meant, should I squash it manually? or is it going to be using the squash & merge from github?

I think what Vince wants is you to collapse all the commits here into a single commit. When the @k8s-ci-robot merges it it will create a merge commit, but I don't think it supports GitHub's "squash-and-merge" feature.

File: bootstrap/kubeadm/controllers/kubeadmconfig_controller.go (hunk 2 above)
Comment (dswij) [reply bootstrap/kubeadm/controllers/kubeadmconfig_controller.go:738]: Thanks, apologies for not reading the logr docs thoroughly.

**Comment by dswij** [comment]
> 
> 
> > > Mind squashing commits?
> > 
> > 
> > no problem.
> > Though, i'm not sure what u meant, should I squash it manually? or is it going to be using the squash & merge from github?
> 
> I think what Vince wants is you to collapse all the commits here into a single commit and force push. When the @k8s-ci-robot merges it it will create a merge commit, but I don't think it supports GitHub's "squash-and-merge" feature.

Right, I was wondering about that (the bot can't squash and merge) as well. I thought it was better to wait for clarification.

**Pull request merged**